packages/lab/src/tabs-next/OverflowMenu.tsx

```diff
@@ -0,0 +1,220 @@
+import { useEffect, useRef, useState } from "react";
+import clsx from "clsx";
+import {
+  autoUpdate,
+  flip,
+  FloatingFocusManager,
+  FloatingPortal,
+  limitShift,
+  shift,
+  size,
+  useClick,
+  useDismiss,
+  useFloating,
+  useInteractions,
+  useListNavigation,
+  useRole,
+} from "@floating-ui/react";
+import { Button, makePrefixer, SaltProvider, Tooltip } from "@salt-ds/core";
+import { OverflowMenuIcon } from "@salt-ds/icons";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+
+import { useOverflowContext, useOverflowMenu } from "@fluentui/react-overflow";
+import { ListItem } from "../list";
+import listCss from "../list/List.css";
+import { isDesktop } from "../window";
+import { TabElement } from "./TabsNextTypes";
+
+const withBaseName = makePrefixer("saltTabstripNext");
+
+type OverflowMenuProps = {
+  tabs: TabElement[];
+  activeTabIndex?: number | null;
+  onSelectItem: (item: { label: string; id: string }) => void;
+  getTabId: (label: string) => string;
+  returnFocusToTabs: () => void;
+};
+
+export function OverflowMenu(props: OverflowMenuProps) {
+  const { isOverflowing } = useOverflowMenu<HTMLButtonElement>();
+
+  if (!isOverflowing) {
+    return null;
+  }
+
+  return <OverflowMenuImpl {...props} />;
+}
+
+function OverflowMenuImpl({
+  tabs,
+  onSelectItem,
+  getTabId,
+  returnFocusToTabs,
+}: OverflowMenuProps) {
+  const { ref, overflowCount } = useOverflowMenu<HTMLDivElement>();
+  const itemVisibility = useOverflowContext(
+    (context) => context.itemVisibility
+  );
+
+  const tabList = tabs
+    .map((tab) => {
+      const label = tab.props.label;
+
+      return {
+        label,
+        id: getTabId(label),
```

Comment:
Yeah, I am currently refactoring the use of ids, so the user can have multiple identical labels, and pass ids. This needs to be documented. Otherwise Overflow will break.